controllers/ingress/eventhandlers/ingress_events.go

```diff
@@ -0,0 +1,74 @@
+package eventhandlers
+
+import (
+	"context"
+	networking "k8s.io/api/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/aws-alb-ingress-controller/pkg/ingress"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var logger = log.Log.WithName("eventhandlers").WithName("ingress")
+
+func NewEnqueueRequestsForIngressEvent(groupLoader ingress.GroupLoader, eventRecorder record.EventRecorder) *enqueueRequestsForIngressEvent {
+	return &enqueueRequestsForIngressEvent{groupLoader: groupLoader, eventRecorder: eventRecorder}
+}
+
+var _ handler.EventHandler = (*enqueueRequestsForIngressEvent)(nil)
+
+type enqueueRequestsForIngressEvent struct {
+	groupLoader   ingress.GroupLoader
+	eventRecorder record.EventRecorder
+}
+
+func (h *enqueueRequestsForIngressEvent) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
+	h.enqueueIfBelongsToGroup(queue, e.Object.(*networking.Ingress))
+}
+
+func (h *enqueueRequestsForIngressEvent) Update(e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
+	ingOld := e.ObjectOld.(*networking.Ingress)
+	ingNew := e.ObjectNew.(*networking.Ingress)
+	if equality.Semantic.DeepEqual(ingOld.Annotations, ingNew.Annotations) &&
+		equality.Semantic.DeepEqual(ingOld.Spec, ingNew.Spec) &&
+		equality.Semantic.DeepEqual(ingOld.DeletionTimestamp.IsZero(), ingNew.DeletionTimestamp.IsZero()) {
+		logger.V(1).Info("ignoring unchanged Ingress Update event", "event", e)
+		return
+	}
```

Comment:
Comment here to explain the changes that we are interested in handling (Annotations, Spec and deletion case with differing timestamp ) would help the reader.